src/main/java/org/candlepin/pki/certs/ContentAccessPayloadKeyBuilder.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2009 - 2025 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.pki.certs;
+
+import org.candlepin.model.Consumer;
+import org.candlepin.model.Environment;
+import org.candlepin.util.Util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HexFormat;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+/**
+ * Builder class responsible for generating a key for a {@link ContentAccessPayload}.
+ */
+public class ContentAccessPayloadKeyBuilder {
+
+    private static final String HASH_ALGORITHM = "SHA-256";
+    private static final int SCA_CONTENT_ACCESS_PAYLOAD_KEY_VERSION = 1;
+    private static final HexFormat HEX_FORMATTER = HexFormat.of();
+
+    private Consumer consumer;
+    private List<Environment> environments = new ArrayList<>();
+
+    /**
+     * Creates a new content access payload builder instance for the provided consumer.
+     *
+     * @param consumer
+     *  the consumer to build a content access payload builder instance for
+     *
+     * @throws IllegalArgumentException
+     *  if the provided consumer is null
+     */
+    public ContentAccessPayloadKeyBuilder(Consumer consumer) {
+        if (consumer == null) {
+            throw new IllegalArgumentException("consumer is null");
+        }
+
+        this.consumer = consumer;
+    }
+
+    /**
+     * Sets the environments used to generate the content access payload key. If a null value is provided then
+     * no environments will be used.
+     *
+     * @param environments
+     *  the environments used to generate the content access payload key
+     *
+     * @return this instance of ContentAccessPayloadKeyBuilder
+     */
+    public ContentAccessPayloadKeyBuilder setEnvironments(List<Environment> environments) {
+        this.environments = environments == null ? new ArrayList<>() : environments;
```

Comment:
If we're going to store an empty list, and we don't expect to be modifying it internally, let's use List.of instead so we aren't actually allocating a new list ever.

Reply:
This is good to know! Updated